brave/brave-core
# Apply device id v2

fix https://github.com/brave/brave-browser/issues/7339

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
### Test SQS Queue migration
1. Establish sync chain between device A and device B without this fix
2. Wait for 2 minutes for SQS queues to be initiated
3. Add A1.com on device A and wait for it appeared on device B
4. Quit Brave on device A
5. Add  B1.com on device B and make sure the record is sent (check log in chrome://inspect/#extensions) and then quit Brave on device B
6. Upgrade Brave on both devices to have this fix
7. device A should be able to receive B1.com shortly
### Test device id v2 migration
1. Establish sync chain between device A and device B without this fix
2. Wait for device list showed on both devices
3. Quit Brave on both devices
4. Upgrade Brave on both devices to have this fix
5. Wait for 1-2 minutes for both devices to send and fetch records (check log in chrome://inspect/#extensions)
6. Delete device B on device A
7. device A should have itself alone and device B is reset
### Test duplicate device id migration
1. Create sync chain on device A and copy the sync code
2. Make device B and device C joined the sync chain at the same time
3. There should be two `(This Device)` on list on device B and C
4. Quit Brave on three devices
5. Upgrade Brave on three devices to have this fix
6. Wait for 1-2 minutes for both devices to send and fetch records (check log in chrome://inspect/#extensions)
7. device list should be normal on three devices
8. Add a bookmark on device A
9. The bookmark should show up on device B and C shortly.
10. Delete device C on device C
11. device A and B should have each other on list and device C is reset
### Old device in sync chain 
1. Establish sync chain between device A and device B without this fix
2. Quit Brave on device A
3. Upgrade device A
4. Add a bookmark on device A
5. device B should have same bookmark shortly
### Old device in sync chain with dup device id
1. Create sync chain on device A and copy the sync code
2. Make device B and device C joined the sync chain at the same time
3. There should be two `(This Device)` on list on device B and C
4. Quit Brave on device A and B
5. Upgrade Brave on device A and device B to have this fix
6. Wait for 1-2 minutes for both devices to send and fetch records (check log in chrome://inspect/#extensions)
7. device list should be normal on device A & B
8. device C should be reset

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
